Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -4666,22 +4666,34 @@ Color LocalFrameView::documentBackgroundColor() const
 
     auto* htmlElement = backgroundDocument->documentElement();
     auto* bodyElement = backgroundDocument->bodyOrFrameset();
-#if ENABLE(FULLSCREEN_API)
-    auto* fullscreenElement = backgroundDocument->fullscreenManager().fullscreenElement();
-#else
-    Element* fullscreenElement = nullptr;
-#endif
 
     // Start with invalid colors.
     Color htmlBackgroundColor;
     Color bodyBackgroundColor;
-    Color fullscreenBackgroundColor;
     if (htmlElement && htmlElement->renderer())
         htmlBackgroundColor = htmlElement->renderer()->style().visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
     if (bodyElement && bodyElement->renderer())
         bodyBackgroundColor = bodyElement->renderer()->style().visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
-    if (fullscreenElement && fullscreenElement->renderer())
-        fullscreenBackgroundColor = fullscreenElement->renderer()->style().visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
+
+#if ENABLE(FULLSCREEN_API)
+    Color fullscreenBackgroundColor = [&] () -> Color {
+        auto* fullscreenElement = backgroundDocument->fullscreenManager().fullscreenElement();
+        if (!fullscreenElement)
+            return { };
+
+        auto* fullscreenRenderer = fullscreenElement->renderer();
+        if (!fullscreenRenderer)
+            return { };
+
+        auto fullscreenElementColor = fullscreenRenderer->style().visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
+
+        WeakPtr backdropRenderer = fullscreenRenderer->backdropRenderer();
+        if (!backdropRenderer)
+            return fullscreenElementColor;
+
+        auto backdropColor = backdropRenderer->style().visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor);
+        return blendSourceOver(backdropColor, fullscreenElementColor);
```

Comment:
Does it apply the `fullscreenElementColor` over the `backdropColor`? This might be incorrect if the fullscreen element is smaller than the backdrop (which is possible) fwiw.

Reply:
Ugh, I think you're right. It would look really weird if you had [white, [black, [white], black], white] in fullscreen, if the element was smaller than the safe area.